test/sqllogictest/transform/union_cancel.slt

```diff
@@ -227,3 +227,55 @@ NULL NULL
 4 5
 5 5
 5 5
+
+statement ok
+CREATE TABLE init(n int, m int, s string);
+
+# Union branch cancellation should happen inside WMR.
+query T multiline
+EXPLAIN
+WITH MUTUALLY RECURSIVE
```

Comment:
nit: the query is probably a bit more readable if you use `l0`, `l1` instead of `c0`, `c1` names.

Reply:
I'm using `c0`, `c1`, instead of the `l0`, `l1`, ..., to reduce confusion in the situation when some transform later suddenly changes the numbering, and then e.g., `l2` in the query would be `l1` in the plan. With using `c0`, `c1`, ..., it's more apparent that the Ids in the plan are not necessarily the same ids as in the query.